hashring.js

```diff
@@ -210,7 +210,10 @@ Hashring.prototype.lookup = function (key) {
     index = 0
   }
 
-  return this._entries[index].peer
+  if (this._entries[index]) {
+    return this._entries[index].peer
+  }
```

Comment:
In which case `index >= this._entries.length`?  I think it _might happen_, but a comment would be good.